Src/Sarif.PatternMatcher.Sdk/Fingerprint.cs

```diff
@@ -26,25 +27,17 @@ public struct Fingerprint
 
         private const char RightBracketReplacement = '\t';
         private const string HashKey = "7B2FD4B8B55B49428DBFB22C9E61D817";
-
         private const string Base64EncodingSymbolSet =
             "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
 
-        private static readonly HashSet<string> s_emptyDenyList = new HashSet<string>();
+        private static readonly ReadOnlyCollection<string> s_emptyDenyList = new ReadOnlyCollection<string>(Array.Empty<string>());
```

Comment:
>ReadOnlyCollection [](http://example.com/codeflow?start=32&length=18)

Suggest using ImmutableHash<string> instead.

Reply:
I thought about it. but we would need to add a new package dependency. its not a bad thing, but since we have only a few elements, the performance is similar

Reply:
that's true. but now ToString can accept a list with duplicates.

it's fine, i think you made the right call for now, for the reasons you mentioned.